polkadot/node/network/collator-protocol/src/validator_side/collation.rs

```diff
@@ -216,25 +221,86 @@ impl CollationStatus {
 }
 
 /// Information about collations per relay parent.
-#[derive(Default)]
 pub struct Collations {
 	/// What is the current status in regards to a collation for this relay parent?
 	pub status: CollationStatus,
 	/// Collator we're fetching from, optionally which candidate was requested.
 	///
-	/// This is the currently last started fetch, which did not exceed `MAX_UNSHARED_DOWNLOAD_TIME`
-	/// yet.
+	/// This is the last fetch for the relay parent. The value is used in
+	/// `get_next_collation_to_fetch` (called from `dequeue_next_collation_and_fetch`) to determine
+	/// if the last fetched collation is the same as the one which just finished. If yes - another
+	/// collation should be fetched. If not - another fetch was already initiated and
+	/// `get_next_collation_to_fetch` will do nothing.
+	///
+	/// For the reasons above this value is not set to `None` when the fetch is done! Don't use it
+	/// to check if there is a pending fetch.
```

Comment:
The name is is confusing. I wanted to to use this field at some point and was surprised by the behaviour so I left a comment with my findings. I agree it would have been better to rename the field and leave a better comment. I'll fix it.